### PR TITLE
docs(common): Add common Data Binding section

### PR DESCRIPTION
**Pull request by dimodi**
This PR includes 2 new common data binding articles - one for Overview + `Data` and one for `OnRead`.

I also included some links to these new articles on other Getting Started pages and data-binding-related articles.

In a next iteration, I will polish the data binding articles of all components to integrate better with these new pages.

### CODE REVIEW

**Comment by marin-bratanov** [comment]
Consider linking this https://docs.telerik.com/blazor-ui/getting-started/value-vs-data-binding from the one that explains the basics of data binding

**Comment by dimodi** [comment]
> Consider linking this https://docs.telerik.com/blazor-ui/getting-started/value-vs-data-binding from the one that explains the basics of data binding

It already does - first paragraph :)

**Comment by svdimitr** [comment]
I love the new article, the explanations of the benefits of the OnRead, the comparison between on read and data, the event agreement section, and the ToDataSourceResult! Pure gold!

**Review by svdimitr — APPROVED** [review]

File: common-features/data-binding/onread.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+### Performance
+
+Large amounts of data require loading in chunks and on demand. This improves the performance of the database, backend, network, and the browser. When a component fires `OnRead`, it expects to receive **only the data items to render**. The exact number depends on the component's `PageSize` parameter.
+
+`OnRead` also offloads data operations outside the component, for example on the remote server. This can improve WebAssembly application performance.
+
+### Customization
+
+`OnRead` allows full control over the data operations. For example, it is possible to use custom sorting and filtering algoritms, if the built-in ones do not fit a given scenario. Here are just a few examples, but there are many more possible scenarios:
```
Comment (dimodi) [inline common-features/data-binding/onread.md:37]:
```suggestion
`OnRead` allows full control over the data operations. For example, it is possible to use custom sorting and filtering algorithms, if the built-in ones do not fit a given scenario. Here are just a few examples, but there are many more possible scenarios:
```

File: common-features/data-binding/overview.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+Some components handle properties with specific names in a predefined way. For example, the Menu will automatically render the `Text` property of its items. If the property names are different, the component will expect additional configuration. All these specifics are described in each component documentation.
+
+
+## Refresh Data
+
+There are three ways to refresh the component data:
+
+* [Bind the component to Observable data]({%slug common-features-observable-data%}). **This option applies only** if the `Data` parameter is set. The component will refresh automatically when items are **added or removed**.
+* Call the component's `Rebind()` method, [if the component has it]({%slug common-features-data-binding-onread%}#components-with-onread-event). This method was implemented to be [used together with the `OnRead` event]({%slug common-features-data-binding-onread%}#refresh-data), but it works with `Data` too. Future UI for Blazor versions will expose the `Rebind()` method for all databound components.
+* Reset the `Data` parameter reference. In some specific scenarios, you may also need to call `StateHasChanged()`.
```
Comment (dimodi) [inline common-features/data-binding/overview.md:99]:
```suggestion
* Reset the `Data` parameter reference. Sometimes, you may also need to call `StateHasChanged()` - for example, if the refreshing occurs in `OnAfterRenderAsync`.
```

File: common-features/data-binding/overview.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        var newId = GridData.Count + 1;
+
+        GridData.FirstOrDefault().Text = DateTime.Now.Ticks.ToString();
+
+        GridData.Add(new SampleModel() {
+            Id = newId,
+            Text = "Text " + newId
+        });
+
+        // components with an OnRead event have Rebind method:
```
Comment (dimodi) [inline common-features/data-binding/overview.md:142]:
```suggestion
        // components with an OnRead event have a Rebind method:
```

File: common-features/data-binding/onread.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+`OnRead` allows full control over the data operations. For example, it is possible to use custom sorting and filtering algorithms, if the built-in ones do not fit a given scenario. Here are just a few examples, but there are many more possible scenarios:
+
+* [Search by multiple data fields in ComboBox and DropDownList]({%slug dropdowns-kb-search-in-multiple-fields%})
+* [Search in hidden Grid columns]({%slug grid-kb-search-in-hidden-fields%})
+* [Debounce Grid data requests]({%slug grid-kb-debounce-operations%})
+* [Debounce ComboBox filter requests]({%slug combo-kb-debounce-onread%})
+
+`OnRead` enables [data binding to **OData** services]({%slug common-kb-odata%}).
+
+In general, `OnRead` allows the application to know the exact data items, which the user is currently seeing.
```
Comment (dimodi) [inline common-features/data-binding/onread.md:46]:
```suggestion
`OnRead` also allows the application to know the exact data items, which the user is currently seeing.
```

File: common-features/data-binding/onread.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+| [MultiSelect]({%slug multiselect-events%}#onread) | - | [MultiSelect virtualization]({%slug multiselect-virtualization%}) |
+
+Components like the [**TreeList**]({%slug treelist-data-binding-load-on-demand%}) and the [**TreeView**]({%slug components/treeview/data-binding/load-on-demand%}) don't have an `OnRead` event. Instead, they load data on demand via `OnExpand` events.
+
+
+## Event Argument
+
+The `OnRead` event handler receives an argument, which inherits from [`ReadEventArgs`](/blazor-ui/api/Telerik.Blazor.Components.ReadEventArgs). The exact type depends on the component. For example, the Grid handler receives `GridReadEventArgs`. The ComboBox handler receives `ComboBoxReadEventArgs`, and so on.
+
+The event argument object has the following properties:
```
Comment (dimodi) [inline common-features/data-binding/onread.md:71]:
```suggestion
The event argument object has the following properties, which are common for all [applicable components](#components-with-onread-event):
```

**Pull request merged**